src/mlpack/methods/cf/cf_impl.hpp

```diff
@@ -326,12 +351,27 @@ void CFType<NormalizationType>::Predict(const arma::Mat<size_t>& combinations,
   for (size_t i = 0; i < queries.n_cols; ++i)
     queries.col(i) = stretchedH.col(users[i]);
 
-  // Now calculate the neighborhood of these users.
-  neighbor::KNN a(stretchedH);
-  arma::mat distances;
+  // Temporary storage for neighborhood of the queried users.
   arma::Mat<size_t> neighborhood;
 
-  a.Search(queries, numUsersForSimilarity, neighborhood, distances);
+  // Now calculate the neighborhood of these users.
+  NeighborSearchPolicy neighborSearch(stretchedH);
+  arma::mat similarities; // Resulting similarities.
+
+  neighborSearch.Search(
+      queries, numUsersForSimilarity, neighborhood, similarities);
+
+  arma::mat weights(numUsersForSimilarity, users.n_elem);
+
+  // Calculate interpolation weights.
+  InterpolationPolicy interpolation(cleanedData);
+  for (size_t i = 0; i < users.n_elem; i++)
+  {
+    arma::vec weightVec;
+    interpolation.GetWeights(weightVec, w, h, users[i],
```

Comment:
This generates an error:
```
candidate function [with VectorType = arma::subview_col<double>] not viable:
expects an l-value for 1st argument
void GetWeights(VectorType& weights,
```
So I think this doesn't work because `weights.col(i)` returns a r-value.

Reply:
Hmm... In that case
```
template<typename VectorType>
void GetWeights(VectorType&& weights, /* other arguments */)
```
should work very well.

Reply:
It works, thanks! As we cannot use `set_size()` for `arma::subview_col<double>` object, I removed `weights.set_size(...)` from the body of `GetWeights()`, and set the size of weights before calling `GetWeights()`. Comments reminding user to initialize the size of weights are also added.